x/incentive/keeper/rewards_swap.go

```diff
@@ -92,5 +98,24 @@ func (k Keeper) SynchronizeSwapReward(ctx sdk.Context, poolID string, owner sdk.
 	claim.Reward = claim.Reward.Add(newRewards...)
 	claim.RewardIndexes = claim.RewardIndexes.With(poolID, globalRewardIndexes)
 
-	k.SetSwapClaim(ctx, claim)
+	return claim
+}
+
+// GetSynchronizedSwapClaim fetches a swap claim from the store and syncs rewards for all pools.
+func (k Keeper) GetSynchronizedSwapClaim(ctx sdk.Context, owner sdk.AccAddress) (types.SwapClaim, bool) {
```

Comment:
+1 this is way cleaner than the previous sync methods, should refactor the others as well when we have time.

Reply:
Hoping to get to it before code freeze